nntrainer/app_context.cpp

```diff
@@ -199,4 +306,31 @@ int AppContext::registerLayerPlugin(const std::string &path,
   return registerFactory<ml::train::Layer>(factory_func, type);
 }
 
+std::vector<int>
+AppContext::registerLayerFromDirectory(const std::string &base_path) {
+  DIR *dir = opendir(base_path.c_str());
+
+  NNTR_THROW_IF(dir == nullptr, std::invalid_argument)
+    << func_tag << "failed to open the directory: " << base_path;
+
+  struct dirent *entry;
+
+  std::vector<int> keys;
+  while ((entry = readdir(dir)) != NULL) {
+    if (endswith(entry->d_name, layerlib_suffix)) {
+      try {
+        int key = registerLayer(entry->d_name, base_path);
+        keys.emplace_back(key);
+      } catch (std::exception &e) {
+        closedir(dir);
+        throw;
```

Comment:
throw e?

Reply:
`throw` is used to rethrow original throw object without loss of information.

refering to https://stackoverflow.com/questions/2360597/c-exceptions-questions-on-rethrow-of-original-exception